pygments/lexers/mcfunction.py

```diff
@@ -0,0 +1,272 @@
+        ],
+
+        "literals": [
+            (r"(true|false)", Keyword.Constant),
+            (r"(-?\d*\.\d+[fFdD]?)", Number.Float),
+            (r"(-?\d+[bBsSlLfFdD]?)", Number.Integer),
+            (r"(-?\d+[eE]-?\d+)", Number.Float),
+            # Seperate states for both types of strings so they don't entangle
+            (r'"', String.Double, "literals.string_double"),
+            (r"'", String.Single, "literals.string_single"),
+        ],
+        "literals.string_double": [
+            (r"\\.", String.Escape),
+            (r'[^\\"\n]+', String.Double),
+            (r'"', String.Double, "#pop"),
+            (r"[$\n]", Error, "#pop"),
+        ],
+        "literals.string_single": [
+            (r"\\.", String.Escape),
+            (r"[^\\'\n]+", String.Single),
+            (r"'", String.Single, "#pop"),
+            (r"[$\n]", Error, "#pop"),
+        ],
+
+        "compound": [
+            # this handles the unquoted snbt keys
+            #  note: stringified keys still work
+            (r"[A-z_]+", Name.Attribute),
+            include("operators"),
+            include("whitespace"),
+            include("literals"),
+            (r"\{", Punctuation, "#push"),
+            (r"\[", Punctuation, "list"),
+            (r"\}", Punctuation, "#pop"),
+        ],
+
+        "list": [
+            (r"(?<=\[)[BIL](?:\;)", Keyword.Type),
+            (r"[A-z_]+", Name.Attribute),
+            include("literals"),
+            include("operators"),
+            include("whitespace"),
+            (r"\{", Punctuation, "#push"),
+            (r"\{", Punctuation, "compound"),
+            (r"\]", Punctuation, "#pop"),
+        ],
+    }
+
+
+class MCFunctionLexer(RegexLexer):
+    """Lexer for the mcfunction scripting language used in Minecraft
+
+    Modelled somewhat after the Github mcfunction grammar:
+    - "https://github.com/Arcensoth/language-mcfunction
+
+    .. versionadded:: ?
+    """
+
+    name = "MCFunction"
+    url = "https://minecraft.fandom.com/wiki/Commands"
+    aliases = ["mcfunction", "mcf"]
+    filenames = ["*.mcfunction"]
+    mimetype = ["text/mcfunction"]
+
+    # Used to denotate the start of a block comment, borrowed from Github's mcfunction
+    _block_comment_prefix = "[>!*$]"
+
+    tokens = {
+        "root": [
+            include("names"),
+            include("nbt"),
+            include("comments"),
+            include("literals"),
+            include("whitespace"),
+            include("operators"),
+            include("delimiters"),
+            include("selectors"),
+        ],
+
+        "names": [
+            # The start of a command (either beginning of line OR after the run keyword)
+            #  We don't encode a list of keywords since mods, plugins, or even pre-processors
+            #  may add new commands, so we have a 'close-enough' regex which catches them.
+            (r"(?=^\s*)([a-z]+)", Name.Builtin),
+            (r"(?<=run\s)([a-z]+)", Name.Builtin),
```

Comment:
Not `\s+`?

Reply:
Ah yes, I should maybe take it out of the group so that It's not included in the look-behind assertion.